src/crawlers/cheerio_crawler.js

```diff
@@ -625,6 +690,8 @@ class CheerioCrawler extends BasicCrawler {
             proxyUrl,
             stream: true,
             useCaseSensitiveHeaders: true,
+            timeoutSecs: this.requestTimeoutMillis / 1000,
+            ...gotoOptions,
```

Comment:
An interesting question whether to merge this at the top or at the bottom. I would merge this at the top because e.g. `stream` cannot be `false`, because it would break the crawler. Also headers should be merged I think, not just overwritten.

Reply:
I would say it should go at the end, or near the end, to allow overriding things - that should be its purpose, or not? Agreed we can move the options that should not be overridable underneath and we can also merge the headers.

Reply:
Yeah you're right that some of the things might make sense to override. We probably need just `stream`. Everything else should work even if overridden. Although overriding a lot of the props could be dangerous and produce weird results. E.g. overriding `url` would cause the crawler to visit a different URL than the `request.url`, which kinda defeats the purpose of having the `Request` instances. But hey, who knows why someone might need to do that.